modules/openxr/extensions/spatial_entities/openxr_spatial_anchor.cpp

```diff
@@ -0,0 +1,1206 @@
+
+OpenXRSpatialAnchorCapability *OpenXRSpatialAnchorCapability::singleton = nullptr;
+
+OpenXRSpatialAnchorCapability *OpenXRSpatialAnchorCapability::get_singleton() {
+	return singleton;
+}
+
+OpenXRSpatialAnchorCapability::OpenXRSpatialAnchorCapability() {
+	singleton = this;
+}
+
+OpenXRSpatialAnchorCapability::~OpenXRSpatialAnchorCapability() {
+	singleton = nullptr;
+}
+
+void OpenXRSpatialAnchorCapability::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("is_spatial_anchor_supported"), &OpenXRSpatialAnchorCapability::is_spatial_anchor_supported);
+	ClassDB::bind_method(D_METHOD("is_spatial_persistence_supported"), &OpenXRSpatialAnchorCapability::is_spatial_persistence_supported);
+
+	ClassDB::bind_method(D_METHOD("is_persistence_scope_supported", "scope"), &OpenXRSpatialAnchorCapability::_is_persistence_scope_supported);
+	ClassDB::bind_method(D_METHOD("create_persistence_context", "scope", "user_callback"), &OpenXRSpatialAnchorCapability::_create_persistence_context, DEFVAL(Callable()));
+	ClassDB::bind_method(D_METHOD("get_persistence_context_handle", "persistence_context"), &OpenXRSpatialAnchorCapability::_get_persistence_context_handle);
+	ClassDB::bind_method(D_METHOD("free_persistence_context", "persistence_context"), &OpenXRSpatialAnchorCapability::free_persistence_context);
+
+	ClassDB::bind_method(D_METHOD("create_new_anchor", "transform", "spatial_context"), &OpenXRSpatialAnchorCapability::create_new_anchor, DEFVAL(RID()));
+	ClassDB::bind_method(D_METHOD("remove_anchor", "anchor_tracker"), &OpenXRSpatialAnchorCapability::remove_anchor);
+	ClassDB::bind_method(D_METHOD("make_anchor_persistent", "anchor_tracker", "persistence_context", "callback"), &OpenXRSpatialAnchorCapability::make_anchor_persistent, DEFVAL(RID()), DEFVAL(Callable()));
+	ClassDB::bind_method(D_METHOD("make_anchor_unpersistent", "anchor_tracker", "persistence_context", "callback"), &OpenXRSpatialAnchorCapability::make_anchor_unpersistent, DEFVAL(RID()), DEFVAL(Callable()));
+
+	ADD_SIGNAL(MethodInfo("persistence_creation_failed", PropertyInfo(Variant::RID, "persistence_context")));
+	ADD_SIGNAL(MethodInfo("anchor_persistent_failed", PropertyInfo(Variant::OBJECT, "tracker", PROPERTY_HINT_RESOURCE_TYPE, "OpenXRAnchorTracker")));
+	ADD_SIGNAL(MethodInfo("anchor_unpersistent_failed", PropertyInfo(Variant::OBJECT, "tracker", PROPERTY_HINT_RESOURCE_TYPE, "OpenXRAnchorTracker")));
+
+	BIND_ENUM_CONSTANT(OPENXR_SPATIAL_PERSISTENCE_SCOPE_SYSTEM_MANAGED);
+	BIND_ENUM_CONSTANT(OPENXR_SPATIAL_PERSISTENCE_SCOPE_LOCAL_ANCHORS);
+}
+
+HashMap<String, bool *> OpenXRSpatialAnchorCapability::get_requested_extensions() {
+	HashMap<String, bool *> request_extensions;
+
+	if (GLOBAL_GET("xr/openxr/extensions/spatial_entity/enabled") && GLOBAL_GET("xr/openxr/extensions/spatial_entity/enable_spatial_anchors")) {
+		request_extensions[XR_EXT_SPATIAL_ANCHOR_EXTENSION_NAME] = &spatial_anchor_ext;
+		if (GLOBAL_GET("xr/openxr/extensions/spatial_entity/enable_persistent_anchors")) {
```

Comment:
Should update instead of `GLOBAL_GET` with `GLOBAL_GET_CACHED` instead.

Reply:
We'll need to do a PR to clean this up in older extensions as well.